packages/gatsby/src/utils/worker/reporter.ts

```diff
@@ -0,0 +1,53 @@
+import reporter from "gatsby-cli/lib/reporter"
+import { ActionCreators } from "gatsby-cli/lib/reporter/types"
+import { GatsbyWorkerPool } from "./types"
+import { isWorker, getMessenger } from "./messaging"
+
+type ReporterActions = typeof ActionCreators
+type ReporterActionKeys = keyof ReporterActions
+
+export function initReporterMessagingInMainProcess(
+  workerPool: GatsbyWorkerPool
+): void {
+  workerPool.onMessage(msg => {
+    if (msg.type === `LOG_INTENT`) {
+      // @ts-ignore Next line (`...msg.payload.args`) cause "A spread argument
+      // must either have a tuple type or be passed to a rest parameter"
+      ActionCreators[msg.payload.name].call(ActionCreators, ...msg.payload.args)
+    }
+  })
+}
+
+const gatsbyWorkerMessenger = getMessenger()
+export function initReporterMessagingInWorker(): void {
+  if (isWorker && gatsbyWorkerMessenger) {
+    if (typeof reporter.setReporterActions === `function`) {
+      const intentifiedActionCreators = (Object.keys(ActionCreators) as Array<
+        ReporterActionKeys
+      >).reduce((acc, keyName: ReporterActionKeys) => {
+        acc[keyName] = (
+          ...args: Parameters<ReporterActions[ReporterActionKeys]>
+        ): void => {
+          gatsbyWorkerMessenger.sendMessage({
+            type: `LOG_INTENT`,
+            payload: {
+              name: keyName,
+              args,
+            } as any,
+          })
+        }
+        return acc
+      }, {}) as ReporterActions
```

Comment:
A bit hard to read with TS juggling. Maybe change it to a simple loop?

Reply:
https://github.com/gatsbyjs/gatsby/pull/32289/commits/9722d61bca86cc115fd51f51fc0a055a6294a0ca is this easier to read now?

Reply:
Reads better 👍